resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/config.scala

```diff
@@ -54,17 +54,17 @@ package object config {
       .stringConf
       .createWithDefault(s"spark-executor:$sparkVersion")
 
-  private[spark] val KUBERNETES_CA_CERT_FILE =
-    ConfigBuilder("spark.kubernetes.submit.caCertFile")
+  private[spark] val KUBERNETES_APISERVER_CA_CERT_FILE =
+    ConfigBuilder("spark.kubernetes.apiserver.caCertFile")
       .doc("""
           | CA cert file for connecting to Kubernetes over SSL. This
           | file should be located on the submitting machine's disk.
         """.stripMargin)
       .stringConf
       .createOptional
 
-  private[spark] val KUBERNETES_CLIENT_KEY_FILE =
-    ConfigBuilder("spark.kubernetes.submit.clientKeyFile")
+  private[spark] val KUBERNETES_CLIENT_APISERVER_KEY_FILE =
+    ConfigBuilder("spark.kubernetes.apiserver.clientKeyFile")
```

Comment:
I think therefore there's a difference we need to make between the fact that the clientKeyFile for the API server is for _authentication_, while the clientKeyFile for the driver submit server is for `TLS`. So we should have the following:

- `spark.ssl.kubernetes.driversubmitserver.keyPem` for the submission server TLS key file,
- `spark.kubernetes.authentication.submission.clientKeyFile` for the key file the submitter uses to contact API server, and
- `spark.kubernetes.authentication.driver.clientKeyFile` for the key file the driver uses to get more executors